spec/util/object_storage/miq_swift_storage_spec.rb

```diff
@@ -0,0 +1,68 @@
+require "util/object_storage/miq_swift_storage"
+
+describe MiqSwiftStorage do
+  context "public methods" do
+    let(:uri) { "swift://foo.com/abc/def" }
+    let(:object_storage) { described_class.new(:uri => uri, :username => 'user', :password => 'pass', :region => 'region') }
+
+    it "#initialize sets the container_name" do
+      container_name = object_storage.container_name
+      expect(container_name).to eq("abc/def")
+    end
+
+    it "#uri_to_object_path returns a new object path" do
+      result = object_storage.uri_to_object_path(uri)
+      expect(result).to eq("def")
+    end
+  end
+
+  describe "#auth_url (private)" do
+    context "with non-ssl security protocol" do
+      let(:uri) { "swift://foo.com:5678/abc/def?region=region&api_version=v3&security_protocol=non-ssl" }
+      let(:object_storage) { described_class.new(:uri => uri, :username => 'user', :password => 'pass') }
+      let(:auth_url) { object_storage.send(:auth_url) }
```

Comment:
This isn't really a good use of `let`. Since this is the actual method under test I think it would be better to call it directly in the test.

Reply:
Got it.  Amended and Pushed.  All comments addressed (again) @NickLaMuro @carbonin